walk-in-interview/src/main/webapp/update-job/script.js

```diff
@@ -0,0 +1,420 @@
+    const requirementsMap = getRequirementsList();
+    const requirementsListElement = document.getElementById("requirements-list");
+
+    // Resets the list in case renders the same requirements twice
+    requirementsListElement.innerHTML = "";
+    const requirementElementTemplate = document.getElementById("requirement-element-template");
+
+    for (const key in requirementsMap) {
+        if (!requirementsMap.hasOwnProperty(key)) {
+            continue;
+        }
+
+        const requirementElement = requirementElementTemplate.cloneNode(/* includes child elements */ true);
+        requirementElement.setAttribute("id", key);
+
+        // tick box
+        const checkbox = requirementElement.children[0];
+        checkbox.setAttribute("id", key);
+        checkbox.setAttribute("value", key);
+
+        // If this requirement is one of the criteria for this job box, tick it
+        tickExistingRequirements(requirementsMap[key], requirements, checkbox);
+
+        // text label
+        const label = requirementElement.children[1];
+        label.setAttribute("for", key);
+        label.innerHTML = requirementsMap[key];
+
+        requirementsListElement.appendChild(requirementElement);
+    }
+}
+
+/**
+ * Ticks the existing requirement.
+ * 
+ * @param {String} requirement Requirement localized name.
+ * @param {String} requirements Requirements list of current job post.
+ * @param {HTMLElement} requirementElement HTML checkbox for that requirement
+ */
+function tickExistingRequirements(requirement, requirements, requirementCheckbox) {
+    if (requirements.includes(requirement)) {
+        requirementCheckbox.setAttribute("checked", true);
+    }
+}
+
+/**
+ * Dynamically adds the options for job pay frequency.
+ * 
+ * @param {String} jobPayFrequency Original payment frequency of the current job.
+ */
+function renderJobPayFrequencyOptions(jobPayFrequency) {
+    const jobPaySelectElement = document.getElementById("pay-frequency");
+    jobPaySelectElement.setAttribute("required", true);
+  
+    renderSelectOptions(jobPaySelectElement, STRINGS["pay-frequency"], jobPayFrequency);
+}
+
+/**
+ * Dynamically adds the options for job duration.
+ * 
+ * @param {String} jobDuration Original duration of the current job.
+ */
+function renderJobDurationOptions(jobDuration) {
+    const jobDurationSelect = document.getElementById("duration");
+    jobDurationSelect.setAttribute("required", true);
+  
+    renderSelectOptions(jobDurationSelect, STRINGS["duration"], jobDuration);
+}
+
+/**
+ * Adds the keys and values from the options map to the select element.
+ * Chooses the existing options.
+ * 
+ * @param {Element} selectElement The select element.
+ * @param {Map} options The map of options to be added.
+ * @param {String} existingOption Existing options.
+ */
+function renderSelectOptions(selectElement, options, existingOption) {
+    selectElement.options.length = 0;
+    selectElement.options[0] = new Option("Select", "");
```

Comment:
perhaps this option is not necessary in this case since there will already be another existing option value.